source/Server.cpp

```diff
@@ -34,6 +36,11 @@ Server::~Server()
   // wl_display_destroy(display);
 }
 
+Server &Server::getInstance() noexcept
```

Comment:
Not a huge deal, but I'd probably move this in the .hpp. Easier inlining for the compiler, which means either improved performance or link time.